sample-app/src/components/DropdownSection.tsx

```diff
@@ -59,30 +60,31 @@ export default function DropdownSection({
   }
 
   function handleKeyDown(evt: globalThis.KeyboardEvent) {
-    if (focusStatus === 'active') {
-      if (['ArrowDown', 'ArrowUp', 'ArrowRight', 'ArrowLeft'].includes(evt.key)) {
-        evt.preventDefault();
-      }
-
-      if (evt.key === 'ArrowDown' || evt.key === 'ArrowRight') {
-        incrementOptionFocus();
-      } else if (evt.key === 'ArrowUp' || evt.key === 'ArrowLeft') {
-        decrementOptionFocus();
-      }
+    if (['ArrowDown', 'ArrowUp', 'ArrowRight', 'ArrowLeft'].includes(evt.key)) {
+      evt.preventDefault();
     }
-  }
 
-  useEffect(() => {
-    if (focusStatus === 'active') {
-      onFocusChange(options[focusedOptionIndex].value, `${optionIdPrefix}-${focusedOptionIndex}`);
+    if (evt.key === 'ArrowDown' || evt.key === 'ArrowRight') {
+      incrementOptionFocus();
+    } else if (evt.key === 'ArrowUp' || evt.key === 'ArrowLeft') {
+      decrementOptionFocus();
+    } else if (evt.key === 'Enter') {
+      onSelect(options[focusedOptionIndex].value, focusedOptionIndex);
     }
-  }, [focusStatus, focusedOptionIndex, onFocusChange, optionIdPrefix, options]);
+  }
 
   useEffect(() => {
     if (focusStatus === 'active') {
+      if (!wasActive.current) {
+        onFocusChange(options[focusedOptionIndex].value, `${optionIdPrefix}-${focusedOptionIndex}`);
```

Comment:
maybe a dumb question but why do we need to call `onFocusChange` in here and also in `increment/decrementOptionFocus`?

Reply:
`increment/decrementOptionFocus` take care of changing what option is focused within the section. when the focus changes (and stays within the section), they call `onFocusChange`. but, when a section that used to be unfocused is focused, that is also a change in focus, which is why we need the `onFocusChange` call in the useEffect